src/meoperators.jl

```diff
@@ -188,684 +178,658 @@ For the definitions of the Lyapunov operators see:
 M. Konstantinov, V. Mehrmann, P. Petkov. On properties of Sylvester and Lyapunov
 operators. Linear Algebra and its Applications 312:35–71, 2000.
 """
-function lyapop(A::AbstractMatrix{T}, E::AbstractMatrix{T}; disc = false, her = false) where T
-    GeneralizedLyapunovMap(A, E; disc = disc, her = her)
+function lyapop(A::AbstractMatrix{T1}, E::AbstractMatrix{T2}; disc=false, her=false) where {T1,T2}
+   T = promote_type(T1, T2)
+   GeneralizedLyapunovMap(convert.(AbstractMatrix{T}, (A, E))...; disc=disc, her=her)
```

Comment:
This conversion is a no-op if both types are equal, so there is no need for another "homogeneous-type" method.